src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -725,7 +725,9 @@ void shutdown(Throwable t) {
             closed = true;
         }
         if (Log.errors()) {
-            if (!(t instanceof EOFException) || isActive()) {
+            if (idleConnectionTimeoutEvent != null) {
+                Log.logTrace("idleConnectionTimeout timeout fired, shutting down connection: {0}", t.getMessage());
+            } else if (!(t instanceof EOFException) || isActive()) {
```

Comment:
That doesn't seem right to me. If `idleConnectionTimeoutEvent` is not null it means that we have armed the timeout, it doesn't mean that it has fired. We're not shutting down the connection, the connection is simply idle, and thus we should log exceptions.

Reply:
Good point, the setup here seems that an idleConnectionTimeoutEvent being not null could cause its log to be incorrectly logged (like you said, it doesn't mean it has fired.

I'm thinking maybe here I could include an additional condition in the form of a check to see if the idleConnectionTimeoutEvent has fired or not as well as the null check.

Reply:
With the latest changes, I'm proposing the additional check for the timeout firing as well as the null check. Also logging with logError as supposed to using logTrace as the idleConnectionTimeoutEvent produceces a ConnectTimeoutException